### PR TITLE
Tutorial 2 Fix #787

### LINKED ISSUE #787: Tutorial Docs: Second tutorial Increment struct mismatch
Hi,
Big fan here :) Anchor indeed elevate the ease of use Solana and makes things much more convenient! thank you!

Back to the issue. In the [second tutorial](https://github.com/project-serum/anchor/blob/master/docs/src/tutorials/tutorial-2.md) I found the document referencing the Increment struct as
```
#[derive(Accounts)]
pub struct Increment<'info> {
    #[account(mut, has_one = authority)]
    pub counter: Account<'info, Counter>,
    #[account(signer)]
    pub authority: AccountInfo<'info>,
}
```

However, in the current code it is
```
#[derive(Accounts)]
pub struct Increment<'info> {
    #[account(mut, has_one = authority)]
    pub counter: Account<'info, Counter>,
    pub authority: Signer<'info>,
}
```

So seems like this tutorial needs a revise to update its information.

### PR DESCRIPTION
**Pull request by nicholasoxford**

### CODE REVIEW

File: docs/src/tutorials/tutorial-2.md [hunk 1]
```diff
@@ -9,11 +9,11 @@ For example, you could imagine easily writing a faulty token program that forget
...
+  from the untrusted `&[AccountInfo]` slice given to a Solana program into a validated struct
+  of deserialized account types.
+- [#[account]](https://docs.rs/anchor-lang/latest/anchor_lang/attr.account.html): attribute macro implementing [AccountSerialize](https://docs.rs/anchor-lang/latest/anchor_lang/trait.AccountSerialize.html) and [AccountDeserialize](https://docs.rs/anchor-lang/latest/anchor_lang/trait.AnchorDeserialize.html), automatically prepending a unique 8 byte discriminator to the account array. The discriminator is defined by the first 8 bytes of the `Sha256` hash of the account's Rust identifier--i.e., the struct type name--and ensures no account can be substituted for another.
+- [ProgramAccount](https://docs.rs/anchor-lang/latest/anchor_lang/struct.ProgramAccount.html): a wrapper type for a deserialized account implementing `AccountDeserialize`. Using this type within an `Accounts` struct will ensure the account is **owned** by the currently executing program.
```
Comment (armaniferrante) [inline docs/src/tutorials/tutorial-2.md:16]:
```suggestion
- [Account](https://docs.rs/anchor-lang/latest/anchor_lang/struct.Account.html): a wrapper type for a deserialized account implementing `AccountDeserialize`. Using this type within an `Accounts` struct will ensure the account is **owned** by the address defined by `declare_id!` where the inner account was defined.
```

File: docs/src/tutorials/tutorial-2.md [hunk 2]
```diff
@@ -48,16 +48,15 @@ Let's focus on the `increment` instruction, specifically the `Increment` struct
...
+    pub authority: Signer<'info>,
 }
 ```
 
 Here, several `#[account(..)]` attributes are used.
 
-* `mut`: tells the program to persist all changes to the account.
-* `has_one`: enforces the constraint that `Increment.counter.authority == Increment.authority.key`.
-* `signer`: enforces the constraint that the `authority` account **signed** the transaction.
+- `mut`: tells the program to persist all changes to the account.
```
Comment (armaniferrante) [inline docs/src/tutorials/tutorial-2.md:57]: Is this a functional change or is it just style?

Comment (nicholasoxford) [reply docs/src/tutorials/tutorial-2.md:57]: Thanks for the quick comment. I am a bit confused why my ide made any changes to this file besides 50-52.. Let me know if I should change them back

Comment (armaniferrante) [reply docs/src/tutorials/tutorial-2.md:57]: No problem I was just curious if there was a functional difference. I dont think there is.

**Pull request merged**